pkg/collector/clock.go

```diff
@@ -82,6 +82,10 @@ func (t *fakeTimer) Reset(d time.Duration) bool {
 		fired = false
 		t.targetTime = clock.now.Add(d)
```

Comment:
Not related to this PR, but I'm wondering if we don't have duplicate timer in the clock, maybe we can break here?